packages/neuron-ui/src/components/ApproveMultisigTx/hooks.ts

```diff
@@ -63,56 +64,82 @@ export const useSignAndExport = ({
       type: AppActions.RequestPassword,
       payload: {
         walletID,
-        actionType: onlyNeedOne ? 'send-from-multisig-need-one' : 'send-from-multisig',
+        actionType: 'send-from-multisig',
         multisigConfig,
       },
     })
-  }, [dispatch, walletID, multisigConfig, offlineSignJson.transaction, onlyNeedOne])
+    closeDialog()
+  }, [dispatch, walletID, multisigConfig, offlineSignJson.transaction, closeDialog])
   return signAndExport
 }
 
 export const useSignedStatus = ({
   multisigConfig,
   signatures,
+  addresses,
 }: {
   multisigConfig: MultisigConfig
   signatures?: Signatures
+  addresses: State.Address[]
 }) => {
   const multisigLockHash = useMemo(() => scriptToHash(addressToScript(multisigConfig.fullPayload)), [
     multisigConfig.fullPayload,
   ])
   const multisigBlake160s = useMemo(() => multisigConfig.addresses.map(v => addressToScript(v).args), [
     multisigConfig.addresses,
   ])
-  const [lackOfRCount, lackOfMCount] = useMemo(() => {
-    let [r, signed] = [0, 0]
-    for (let i = 0; i < multisigConfig.r; i++) {
-      if (!signatures?.[multisigLockHash]?.includes(multisigBlake160s[i])) {
-        r = multisigConfig.r - i
+  const addressBlake160s = useMemo(() => addresses.map(v => addressToScript(v.address).args), [addresses])
+  return useMemo(() => {
+    const notSpecifiedCount = multisigConfig.m - multisigConfig.r
+    const specifiedUnsignedAddresses: string[] = []
+    const unspecifiedSignedAddresses: string[] = []
+    const unspecifiedUnsignedAddresses: string[] = []
+    for (let i = 0; i < multisigBlake160s.length; i++) {
+      const hasSigned = signatures?.[multisigLockHash]?.includes(multisigBlake160s[i])
+      if (i < multisigConfig.r) {
+        if (!hasSigned) {
+          specifiedUnsignedAddresses.push(multisigBlake160s[i])
+        }
+      } else {
+        ;(hasSigned ? unspecifiedSignedAddresses : unspecifiedUnsignedAddresses).push(multisigBlake160s[i])
```

Comment:
What does ` ? unspecifiedSignedAddress :` do?

Reply:
This is convenient for calculate lack of unspecified signer. I wrote an extra semicolon, i will delete it.